services/spar/src/Spar/API.hs

```diff
@@ -84,7 +86,10 @@ type API = "i" :> "status" :> Get '[JSON] NoContent
       -- NB. If you add endpoints here, also update Test.Spar.APISpec
 
 type APIMeta     = "sso" :> "metadata" :> SAML.APIMeta
-type APIAuthReq  = "sso" :> "initiate-login" :> SAML.APIAuthReq
+type APIAuthReq  = "sso" :> "initiate-login"
+                :> QueryParam "success_redirect" URI.URI
+                :> QueryParam "error_redirect" URI.URI
+                :> SAML.APIAuthReq
```

Comment:
Why?  In principle GET is not expected to have any effects, but here I would argue that POST is more confusing because it suggests that we are storing something rather than picking something up.

Anyway I think this is an ambiguous case and it's not worth the time debating.  Unless you have any points I am missing?